front/src/applications/operationalStudies/views/Scenario/components/Timetable/utils.ts

```
@@ -70,7 +68,26 @@ export const exportTimetableItems = (
       { train_schedules: [], paced_trains: [] }
     );
 
-  const jsonString = JSON.stringify(formattedTimetableItems);
+  return JSON.stringify(formattedTimetableItems);
+};
+
+export const copyTimetableItemsToClipboard = (
+  selectedTimeTableIdsFromClick: TimetableItemId[],
+  timetableItems: TimetableItem[]
+) => {
+  if (!timetableItems) return;
+
+  const jsonString = timetableItemsToJson(selectedTimeTableIdsFromClick, timetableItems);
+  navigator.clipboard.writeText(jsonString);
```

Could we use `ClipboardItem` + `clipboard.write()` instead of text, to indicate that a JSON blob is being copied? Could also set disposition as attachment.

---

Unfortunately from what I saw, the clipboard doesn't support writing JSON, it can only write plain text, html or png. :(

---

Hm, what do you mean? Does it throw an error? Does it silently do nothing?

---

It throw an error (DOMException with not supported type to be more specific).

---

Ah right, `ClipboardItem.supports('application/json')` returns false… It seems like we'd need to use a [custom `web` MIME type](https://w3c.github.io/clipboard-apis/#optional-data-types-x) if we wanted to assign a different MIME type…

---

@emersion What is the purpose and added value of having a custom MIME type? We have never worked with this so we are not really sure

---

I tried the custom web type, it works on chromium, on firefox it throws an error and on edge you can't paste outside the website haha.

---

The purpose of the non-text/plain MIME type is to be able to reliably tell apart timetable items from regular text when pasting.

But yeah, since it's not supported by browsers, I suppose we don't have a choice here.